tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,399 @@
+"""Tests for the ``subnets`` paths.
+
+An API reference is available here:
+http://theforeman.org/api/apidoc/v2/1.15.html
+
+
+:Requirement: Subnet
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+from robottelo.test import APITestCase
+from robottelo.decorators import stubbed, tier1, tier2, tier3
+from robozilla.decorators import skip_if_bug_open
+
+
+class ParameterizedSubnetTestCase(APITestCase):
+    """Implements parametrized subnet tests in API"""
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter(self):
+        """Subnet with parameters can be created
+
+        :id: c1dae6f4-45b1-45db-8529-d7918e41a99b
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter with single key and single value
+
+        :expectedresults: The parameter should be created in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_multiple_values(self):
+        """Subnet parameters can be created with multiple values
+
+        :id: b3de6f96-7c39-4c44-b91c-a6d141f5dd6a
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having single key and multiple values
+                separated with comma
+
+        :expectedresults: The parameter with multiple values should be saved
+            in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_multiple_names(self):
+        """Subnet parameters can be created with multiple names with valid
+        separators
+
+        :id: d1e2d75a-a1e8-4767-93f1-0bb1b75e10a0
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
```

Comment:
I renamed the test case ... i think its worth testing key name with valid and valid. As that can make program unstable if not handled properly.